OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -709,11 +709,14 @@ static class DreamProcNativeRoot {
         [DreamProcParameter("n", Type = DreamValueType.Float)]
         public static DreamValue NativeProc_fract(NativeProc.State state) {
             DreamValue arg = state.GetArgument(0, "n");
-            if (arg.TryGetValueAsFloat(out float floatnum)) {
-                if(float.IsInfinity(floatnum)) {
+            if (arg.TryGetValueAsFloat(out var floatNum)) {
+                if(float.IsInfinity(floatNum)) {
                     return new DreamValue(0);
                 }
-                return new DreamValue(floatnum - MathF.Truncate(floatnum));
+                if(float.IsNaN(floatNum)) {
+                    return arg;
+                }
+                return new DreamValue(floatNum - (int)floatNum);
```

Comment:
Why should `(int)floatNum` be preferred over `MathF.Truncate(floatNum)`?

Reply:
they do roughly the same things do they not; subtraction shouldnt be a particularity expensive op. Although im not 100% sure on intcasting